Sources/Tests/FeatureTests/pfPasswordStoreTest/test_pfPasswordStore.cpp

```diff
@@ -0,0 +1,130 @@
+containing parts covered by the terms of the Bink SDK EULA, 3ds Max EULA,
+PhysX SDK EULA, DirectX SDK EULA, OpenSSL and SSLeay licenses, IJG
+JPEG Library README, Windows Media SDK EULA, or QuickTime SDK EULA, the
+licensors of this Program grant you additional
+permission to convey the resulting work. Corresponding Source for a
+non-source form of such a combination shall include the source code for
+the parts of OpenSSL and IJG JPEG Library used as well as that of the covered
+work.
+
+You can contact Cyan Worlds, Inc. by email legal@cyan.com
+ or by snail mail at:
+      Cyan Worlds, Inc.
+      14617 N Newport Hwy
+      Mead, WA   99021
+
+*==LICENSE==*/
+
+#include <gtest/gtest.h>
+#include <string_theory/string>
+#include "plFileSystem.h"
+#include "pnUUID/pnUUID.h"
+#include "pfPasswordStore/pfPasswordStore.h"
+#include "pfPasswordStore/pfPasswordStore_impl.h"
+
+TEST(pfPasswordStore, service)
+{
+    pfPasswordStore* store = pfPasswordStore::Instance();
+    EXPECT_NE(store, nullptr);
+
+#if defined(HS_BUILD_FOR_WIN32)
+    EXPECT_EQ(typeid(*store), typeid(pfWin32PasswordStore));
+#elif defined(HS_BUILD_FOR_APPLE)
+    EXPECT_EQ(typeid(*store), typeid(pfApplePasswordStore));
+#elif defined(HAVE_LIBSECRET)
+    EXPECT_EQ(typeid(*store), typeid(pfUnixPasswordStore));
+#else
+    EXPECT_EQ(typeid(*store), typeid(pfFilePasswordStore));
+#endif
+}
+
+#if defined(HS_BUILD_FOR_WIN32)
+TEST(pfWin32PasswordStore, storing_and_retrieval)
+{
+    pfWin32PasswordStore store;
+    ST::string username = "test_pfPasswordStore";
+    ST::string password = plUUID::Generate().AsString();
+
+    bool success = store.SetPassword(username, password);
+    EXPECT_EQ(success, true);
+
+    ST::string pass = store.GetPassword(username);
+    EXPECT_EQ(pass.compare(password), 0);
+}
+#endif
+
+#if defined(HS_BUILD_FOR_APPLE)
+TEST(pfApplePasswordStore, storing_and_retrieval)
+{
+    pfApplePasswordStore store;
+    ST::string username = "test_pfPasswordStore";
+    ST::string password = plUUID::Generate().AsString();
+
+    bool success = store.SetPassword(username, password);
+    EXPECT_EQ(success, true);
+
+    ST::string pass = store.GetPassword(username);
+    EXPECT_EQ(pass.compare(password), 0);
+}
+#endif
+
+#if defined(HAVE_LIBSECRET)
+TEST(pfUnixPasswordStore, storing_and_retrieval)
+{
+    pfUnixPasswordStore store;
+    ST::string username = "test_pfPasswordStore";
+    ST::string password = plUUID::Generate().AsString();
+
+    bool success = store.SetPassword(username, password);
+    EXPECT_EQ(success, true);
+
+    ST::string pass = store.GetPassword(username);
+    EXPECT_EQ(pass.compare(password), 0);
+}
+#endif
+
+TEST(pfFilePasswordStore, storing_and_retrieval)
+{
+    // In internal builds, if a local init/login.dat file exists, it will be
+    // used instead of the one in the User AppData folder. For testing, we'll
+    // ensure that file exists to avoid overwriting real login.dat credentials
+    // in the User AppData folder.
```

Comment:
This smells like a hack...  If we really want to enable testing, we should allow a test override, instead of relying on implementation details that may change in the future.